src/services/lobby.service.js

```diff
@@ -0,0 +1,113 @@
+const { v4: uuidv4 } = require('uuid');
+const Lobby = require('../models/lobby.model');
+const socketService = require('./socket.service');
+const userService = require('./user.service');
+
+const getRandomString = (len) => {
+  const arr = '123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ';
+  let ans = '';
+  for (let i = len; i > 0; i -= 1) {
+    ans
+      += arr[(Math.floor(Math.random() * arr.length))]; // NOSONAR not used in secure contexts
+  }
+  return ans;
+};
+
+const getByLobbyID = async (lobbyid) => Lobby.findOne({ lobbyid });
+
+const isPlayerInLobby = async (uuid) => Lobby.findOne({ players: uuid });
+exports.getAll = async () => Lobby.find({});
+exports.delete = async (uuid) => Lobby.deleteOne({ uuid: uuid.toString() }).then(((data) => {
+  if (data.deletedCount === 0) {
+    throw Error('Lobby not found');
+  }
+  return 'delete successfull';
+}));
+exports.create = async (name, isPublic, maxPlayers, playerUUID) => new Promise(
+  (resolve, reject) => {
+    isPlayerInLobby(playerUUID).then((data) => {
+      if (data != null) {
+        reject(new Error('Player is already in an lobby'));
+      } else {
+        const lobby = new Lobby({
+          uuid: uuidv4(),
+          lobbyid: getRandomString(6),
```

Comment:
We should probably add a check if the id is unique or set `unique: true` in the `lobby.model.js`